sdk/communication/Azure.Communication.Identity/tests/samples/Sample1_CommunicationIdentityClient.cs

```diff
@@ -149,6 +152,36 @@ public async Task CreateIdentityWithAccessKey()
             }
         }
 
+        [Test]
+        public async Task ExchangeAccessToken()
+        {
+            IPublicClientApplication publicClientApplication = PublicClientApplicationBuilder.Create(TestEnvironment.CommunicationM365AppId)
+                                                .WithAuthority(TestEnvironment.CommunicationM365AadAuthority + "/" + TestEnvironment.CommunicationM365AadTenant)
+                                                .WithRedirectUri(TestEnvironment.CommunicationM365RedirectUri)
+                                                .Build();
+            string[] scopes = { TestEnvironment.CommunicationM365Scope };
+            SecureString communicationMsalPassword = new SecureString();
+            foreach (char c in TestEnvironment.CommunicationMsalPassword)
+            {
+                communicationMsalPassword.AppendChar(c);
+            }
+            AuthenticationResult result = await publicClientApplication.AcquireTokenByUsernamePassword(
+                scopes,
+                TestEnvironment.CommunicationMsalUsername,
```

Comment:
How about adding a test for token acquired with a different scope than M365.. so it would still be a valid token but would not work for teams

Reply:
We can't generate a non M365 token unless we get some credentials.